src/gui/fspathedit_p.cpp

```diff
@@ -80,6 +255,33 @@ void Private::FileLineEdit::keyPressEvent(QKeyEvent *e)
         m_completerModel->setRootPath(QFileInfo(text()).absoluteDir().absolutePath());
         showCompletionPopup();
     }
+
+    const FileSystemPathValidator *validator =
+        dynamic_cast<const FileSystemPathValidator *>(this->validator());
```

Comment:
FileSystemPathValidator does not contain Q_OBJECT, thus qobject_cast would not work.